Diffs with V8

mtrofin opened the pull request:
Added more tests.

Made changes that illustrate differences with V8.

bnjbvr committed:
Primary infrastructure for stage testing:

- can compile wast to JS file
- can convert JS files to WPT test cases
- can create a front page with all the JS tests

bnjbvr committed:
Build;

bnjbvr committed:
Fix relative URL;

mtrofin committed:
More tests + conformance differences.

mtrofin committed:
use promise_test in the async cases

mtrofin committed:
WebAssembly.instantiate return pair property attributes

rossberg committed:
Fix JS assert_soft_invalid

mtrofin committed:
Removed V8 commented out stuff

rossberg committed:
Make JS harness optional and compatible with async (#407)

rossberg committed:
Tests for checking segments against actual table/memory sizes (#408)

bnjbvr commented on test/js-api/jsapi.js:
```diff
@@ -683,48 +710,75 @@ test(() => {
     const instantiate = WebAssembly.instantiate;
     assert_equals(instantiate, instantiateDesc.value);
-    assert_equals(instantiate.length, 2);
+    assert_equals(instantiate.length, 1);
```
Per ecma262, `Every built-in Function object, including constructors, has a length property whose value is an integer. Unless otherwise specified, this value is equal to the largest number of named arguments shown in the subclause headings for the function description.`, so it seems 2 was correct for the function can take up to 2 arguments `instantiate(BufferSource bytes [, importObject])`.

rossberg replied:
Note that Ecma 262 Sec. 17 continues saying "Optional parameters (which are indicated with brackets:  [ ]) or rest parameters (which are shown using the form «...name») are not included in the default argument count.". So also the following Note 2 for additional clarification. So we should only count `bytes`.

mtrofin replied:
Left to "1"

bnjbvr commented on test/js-api/jsapi.js:
```diff
@@ -357,12 +375,12 @@ test(() => {
     assert_equals(typeof exportsObj, "object");
     assert_equals(Object.isExtensible(exportsObj), false);
-    assert_equals(Object.getPrototypeOf(exportsObj), null);
+    assert_equals(Object.getPrototypeOf(exportsObj), Object.prototype);
```
Can you revert this chunk?

mtrofin replied:
Done

bnjbvr commented on test/js-api/jsapi.js:
```diff
@@ -357,12 +375,12 @@ test(() => {
     assert_equals(Object.keys(exportsObj).join(), "f");
     assertErrorMessage(() => Object.setPrototypeOf(exportsObj, {}), TypeError, /can't set prototype of this object/);
-    assert_equals(Object.getPrototypeOf(exportsObj), null);
+    assert_equals(Object.getPrototypeOf(exportsObj), Object.prototype);
```
And this one?

mtrofin replied:
Done

bnjbvr reviewed (changes requested):
Thanks for the PR! I'll include it in the PR to the spec repo.

bnjbvr committed:
Add an HTML test case;

HTML test cases (in opposition to pure JS test cases) are JS test cases that
need a full browser (DOM) environment. For the sake of example, here is a test
for indexeddb loading and storing a WebAssembly module.

bnjbvr committed:
Build;

mtrofin committed:
exports.prototype is null

mtrofin committed:
exports.prototype is null

mtrofin committed:
exports.prototype is null, and instantiate.length is 2

mtrofin committed:
Merge branch 'mtrofin' of github.com:mtrofin/wasm-spec into mtrofin

mtrofin commented:
ptal

bnjbvr commented:
Thanks! I've force-pushed to my repo (to rebase over upstream changes), so I'll cherry-pick your commits and integrate them in my PR.

mtrofin committed:
Merge branch 'gh-pages' into mtrofin

mtrofin commented:
Awesome, thanks - let me know when I can close this, then.

bnjbvr commented:
Cherry-picked as many commits, thanks.
